ui/components/multichain-accounts/address-qr-code-modal/address-qr-code-modal.tsx

```diff
@@ -0,0 +1,249 @@
+import type { ModalProps } from '../../component-library';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { useCopyToClipboard } from '../../../hooks/useCopyToClipboard';
+import { openBlockExplorer } from '../../multichain/menu-items/view-explorer-menu-item';
+import { MetaMetricsContext } from '../../../contexts/metametrics';
+
+// Constants for QR code generation
+const QR_CODE_TYPE_NUMBER = 4;
+const QR_CODE_CELL_SIZE = 5;
+const QR_CODE_MARGIN = 16;
+const QR_CODE_ERROR_CORRECTION_LEVEL = 'M';
+
+// Constants for address segmentation
+const PREFIX_LEN = 6;
+const SUFFIX_LEN = 5;
+
+export type AddressQRCodeModalProps = Omit<
+  ModalProps,
+  'isOpen' | 'onClose' | 'children'
+> & {
+  isOpen: boolean;
+  onClose: () => void;
+  address: string;
+  accountName: string;
+  networkName: string;
+  networkImageSrc?: string | undefined;
+};
+
+export const AddressQRCodeModal: React.FC<AddressQRCodeModalProps> = ({
+  isOpen,
+  onClose,
+  address,
+  accountName,
+  networkName,
+  networkImageSrc,
+}) => {
+  const t = useI18nContext();
+  const [copied, handleCopy] = useCopyToClipboard();
+  const trackEvent = useContext(MetaMetricsContext);
+
+  // Address segmentation for display
+  const addressStart = address.substring(0, PREFIX_LEN);
+  const addressMiddle = address.substring(
+    PREFIX_LEN,
+    address.length - SUFFIX_LEN,
+  );
+  const addressEnd = address.substring(address.length - SUFFIX_LEN);
+
+  // Generate QR code
+  const qrImage = qrCode(QR_CODE_TYPE_NUMBER, QR_CODE_ERROR_CORRECTION_LEVEL);
+  qrImage.addData(address);
+  qrImage.make();
+
+  // Handle copy address
+  const handleCopyClick = useCallback(() => {
+    handleCopy(address);
+  }, [address, handleCopy]);
+
+  // TODO: Move this out into a utility or selector
+  // Centralized explorer configuration
+  const explorerInfo = useMemo(() => {
+    const networkNameLower = networkName.toLowerCase();
+
+    if (networkNameLower.includes('ethereum')) {
+      return {
+        url: 'https://etherscan.io',
+        name: 'Etherscan',
+        buttonText: t('viewAddressOnExplorer', ['Etherscan']),
+      };
+    }
+
+    if (networkNameLower.includes('solana')) {
+      return {
+        url: 'https://solscan.io',
+        name: 'Solscan',
+        buttonText: t('viewAddressOnExplorer', ['Solscan']),
+      };
+    }
+
+    if (networkNameLower.includes('bitcoin')) {
+      return {
+        url: 'https://blockstream.info',
+        name: 'Blockstream',
+        buttonText: t('viewAddressOnExplorer', ['Blockstream']),
+      };
+    }
+
+    // Return null if no valid explorer found - button won't be shown
+    return null;
+  }, [networkName, t]);
+
+  const handleExplorerNavigation = useCallback(() => {
+    if (!explorerInfo) {
+      return;
+    }
+
+    const addressLink = `${explorerInfo.url}/address/${address}`;
+    openBlockExplorer(addressLink, 'Address QR Code Modal', trackEvent);
+  }, [address, explorerInfo, trackEvent]);
```

Comment:
This explorer function isn't complete and I've opted to hide it when the network doesn't exist. It is by no means an exhaustive list and should be improved in a subsequent PR that is out of the scope of this one.